### DISCUSSION TITLE
ValidationError when extracting children of a block

### DISCUSSION THREAD

**Discussion by dylanjcastillo** [ramnes/notion-sdk-py#178] [Q&A] [answered]
Hi there,

I'm getting a `ValidationError` when trying to extract the children of a block using `notion.blocks.children.list("XYZ")`. This is the error message:

```
Couldn't get children for block: XYZ, (child_page): 1 validation error for BlockList
results -> 16
```

The block contains paragraphs, formulas, tables, code, images, and headings. Interestingly, `notion.blocks.retrieve("XYZ")` works correctly, so the issue seems to be specific to the children generator.

Have you come across this issue before?

**Comment by ramnes** [comment, marked as answer]
Edit: Answer in comments. The error was raised by notional, not notion-sdk-py.

Hey!

I'm converting this in a discussion, because notion-sdk-py doesn't raise ValidationError by itself since it doesn't do any validation, so it's quite unlikely that the issue is on our side.

Please edit your post with a complete stack trace and more code and I'll be happy to take a look. :)

**Reply by dylanjcastillo** [reply]
Hey @ramnes, thank you for the quick reply. Here's the complete stack trace:

```
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
Cell In[23], line 3
      1 page_id = "XYZ"
----> 3 result = extract_from_page(page_id)

Cell In[22], line 34, in extract_from_page(page_id)
     31 output = {"title": title } | get_metadata(page) | {"blocks": []}
     33 for block in notion.blocks.children.list(page_id):
---> 34     block_output = extract_from_block(block)
     35     if not block_output:
     36         continue

Cell In[22], line 14, in extract_from_block(block)
     12 if block.has_children:
     13     block_output["blocks"] = []
---> 14     for child in notion.blocks.children.list(block):
     15         processed_child = extract_from_block(child)
     16         if not processed_child: 

File ~/Documents/GitHub/test.venv/lib/python3.10/site-packages/notional/iterator.py:149, in EndpointIterator.__call__(self, **kwargs)
    145 self.page_num += 1
    147 page = self._endpoint(start_cursor=self.next_cursor, **kwargs)
--> 149 api_list = ObjectList.parse_obj(page)
    151 for obj in api_list.results:
...
File ~/Documents/GitHub/test/.venv/lib/python3.10/site-packages/pydantic/main.py:342, in pydantic.main.BaseModel.__init__()

ValidationError: 1 validation error for BlockList
results -> 16
  Unsupported sub-type: equation (type=type_error)
```

Looks like an issue with `equation`?

**Reply by ramnes** [reply]
The error is raised by notional, not notion-sdk-py. :)

**Reply by dylanjcastillo** [reply]
Oh, you're completely correct. Sorry about that!
